models/contacts.js

```diff
@@ -1,14 +1,97 @@
-// const fs = require('fs/promises')
+const fs = require('fs/promises');
+const path = require('path');
+const Joi = require("joi");
+const { v4: uuidv4 } = require("uuid");
 
-const listContacts = async () => {}
+console.log("here is my language")
 
-const getContactById = async (contactId) => {}
+const contactsPath = path.format({ dir: "./models", base: "contacts.json"});
 
-const removeContact = async (contactId) => {}
 
-const addContact = async (body) => {}
+const listContacts = async (filePath, arr) => {
+  const string = JSON.stringify(arr, null, 2);
+  try {
+    await fs.writeFile(filePath, string);
+  } catch (err) {
+    console.error(err)
+  }
+};
 
-const updateContact = async (contactId, body) => {}
+const getContactById = async (contactId) => {
+  const contactData = await listContacts();
```

Comment:
Upewnij się zawsze że otrzymujesz to co chcesz. Jeśli w tej listContacts funkcji będzie błąd wówczas w bloku catch zostanie on wysłany do konsoli. I ta funkcja listContacts zwróci undefined. Wtedy wywołanie funkcji w linijce 22 filter na undefined spowoduje błąd.